src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,488 @@
+      make_not_null(&char_projected_rhs_dt_v_zero),
+      make_not_null(&char_projected_rhs_dt_v_minus),
+      make_not_null(&constraint_char_zero_plus),
+      make_not_null(&constraint_char_zero_minus), make_not_null(&char_speeds),
+      face_mesh_velocity, normal_covector, pi, phi, spacetime_metric, coords,
+      gamma1, gamma2, lapse, gauge_source, spacetime_deriv_gauge_source, dt_pi,
+      dt_phi, dt_spacetime_metric, d_pi, d_phi, d_spacetime_metric);
+
+  // Account for moving mesh: char speeds -> cher speeds - n_i v^i_g
+  if (face_mesh_velocity.has_value()) {
+    const auto negative_lambda0 =
+        get(dot_product(normal_covector, *face_mesh_velocity));
+    for (size_t a = 0; a < 4; ++a) {
+      char_speeds.at(a) -= negative_lambda0;
+    }
+  }
+
+  detail::set_dt_v_psi_constraint_preserving(
+      make_not_null(&bc_dt_v_psi), unit_interface_normal_vector,
+      three_index_constraint, char_projected_rhs_dt_v_psi, char_speeds);
+
+  detail::set_dt_v_zero_constraint_preserving(
+      make_not_null(&bc_dt_v_zero), unit_interface_normal_vector,
+      four_index_constraint, char_projected_rhs_dt_v_zero, char_speeds);
+
+  if (type_ == detail::ConstraintPreservingBjorhusType::ConstraintPreserving) {
+    detail::set_dt_v_minus_constraint_preserving(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, incoming_null_one_form,
+        outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+        projection_ab, projection_Ab, projection_AB,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, char_speeds);
+  } else if (type_ == detail::ConstraintPreservingBjorhusType::
+                          ConstraintPreservingPhysical) {
+    detail::set_dt_v_minus_constraint_preserving_physical(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, normal_covector,
+        unit_interface_normal_vector, spacetime_unit_normal_vector,
+        incoming_null_one_form, outgoing_null_one_form, incoming_null_vector,
+        outgoing_null_vector, projection_ab, projection_Ab, projection_AB,
+        inverse_spatial_metric, extrinsic_curvature, spacetime_metric,
+        inverse_spacetime_metric, three_index_constraint,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, phi, d_phi, d_pi,
+        char_speeds);
+  } else {
+    ERROR("Failed to set dtVMinus. Input option \""
+          << "\" Must be one of ConstraintPreserving or "
+             "ConstraintPreservingPhysical");
+  }
+
+  // Only add corrections at grid points where the char speeds are negative
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_psi), char_speeds.at(0));
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_zero), char_speeds.at(1));
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
```

Comment:
Is there some mathematical guarantee that means that v_plus doesn't need this check also?

Reply:
The characteristic speed of `v_plus` is `lapse - normal _k shift^k`. The hope here is that the external boundary is sufficiently far away that `lapse --> 1` and  `shift^k --> 0`, and also the mesh velocity is not close to being luminal. This would fail if, of course, the boundary is very close to the source, but then so will this whole treatment of the boundary conditions anyway.